lib/experimental/new_embed.dart

```diff
@@ -674,12 +700,41 @@ class NewEmbed {
 // toggle that class.
 class NewEmbedTabController extends TabController {
   final MDCTabBar _tabBar;
+  bool _userHasSeenSolution = false;
 
   NewEmbedTabController(this._tabBar);
 
+  void registerTab(TabElement tab) {
+    tabs.add(tab);
+
+    try {
+      tab.onClick
+          .listen((_) => selectTab(tab.name, force: _userHasSeenSolution));
+    } catch (e, st) {
+      print('Error from registerTab: $e\n$st');
+    }
+  }
+
   /// This method will throw if the tabName is not the name of a current tab.
   @override
-  void selectTab(String tabName) {
+  Future selectTab(String tabName, {bool force = false}) async {
```

Comment:
Do you anticipate anyone calling this with `force`, or would it be cleaner to just check `_userHasSeenSolution` inside `selectTab`?

Reply:
good point, let's remove it for now - it's not clear we're going to need it